cpp/tests/io/parquet_test.cpp

```diff
@@ -4204,4 +4216,47 @@ TEST_F(ParquetReaderTest, StructByteArray)
   CUDF_TEST_EXPECT_TABLES_EQUAL(expected, result.tbl->view());
 }
 
+TEST_P(ParquetSizedTest, DictionaryTest)
+{
+  constexpr int nrows = 3'000'000;
+  char buf[64];
+
+  auto elements       = cudf::detail::make_counting_transform_iterator(0, [&buf](auto i) {
+    sprintf(buf, "a unique string value suffixed with %d", i % GetParam());
```

Comment:
This is fascinating to me and totally a C-ism. I can tell because this looks like something I would have written in my past. :) The thing I don't get is why we take this time to build a static-length buffer, pass it into a lambda, use it semi-unsafely with `sprintf` and then dump it into a `std::string`?

My first thought was is performance really that critical that we can't define buf inside the lambda, but then I went further and said why not build a std::string from the start? Is there a reasoning behind that I am missing?

```
[](auto i) { return "a unique string value suffixed with " + std::to_string(i % GetParam()); }

Reply:
> This is fascinating to me and totally a C-ism. I can tell because this looks like something I would have written in my past. :)

SIgh. Old habits die hard :(  Will fix